lib/tcp.js

```diff
@@ -157,7 +157,10 @@ exports.client = function (options, transportUtil) {
       reconnect.on('disconnect', function (err) {
         seneca.log.debug('client', type, 'disconnect', spec, topic, clientOptions,
            (err && err.stack) || err)
-        _.find(connections, { clientOptions: clientOptions }).setup = false
+        var connections = _.find(connections, { clientOptions: clientOptions });
```

Comment:
This would fix the issue
```
var connection = _.find(connections, {
    clientOptions: clientOptions
})

if (connection !== undefined) {
    connection.setup = false
}
```